components/net/protocols/mod.rs

```diff
@@ -68,7 +68,7 @@ pub trait ProtocolHandler: Send + Sync {
 
 #[derive(Default)]
 pub struct ProtocolRegistry {
-    pub(crate) handlers: HashMap<String, Box<dyn ProtocolHandler>>, // Maps scheme -> handler
+    pub(crate) handlers: FxHashMap<String, Box<dyn ProtocolHandler>>, // Maps scheme -> handler
```

Comment:
This is registerable only via embededer, right? I think it's okay to not use secure hashing, although I am not sure about performance (IIRC fxhash is only good for u64 or less).